src/IIS/Settings/Bindings/CustomBindingSettings.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Cake.IIS.Settings.Bindings.FluentAPI;
+
+namespace Cake.IIS.Settings.Bindings
+{
+    /// <summary>
+    /// Class to represent and configure any type of IIS binding (secure or not).
+    /// </summary>
+    public class CustomBindingSettings : ICustomBindingSettings, ISecureBindingSettings
+    {
+        /// <summary>
+        /// Creates new instance of <see cref="CustomBindingSettings"/>.
+        /// </summary>
+        /// <param name="bindingProtocol">Binding type.</param>
+        public CustomBindingSettings(BindingProtocol bindingProtocol)
+        {
+            this.BindingProtocol = bindingProtocol;
+        }
+
+        /// <summary>
+        /// Gets or sets IP Address
+        /// </summary>
+        public string IpAddress { get; set; }
+
+        /// <summary>
+        /// Gets or sets IP Port
+        /// </summary>
+        public int Port { get; set; }
+
+        /// <summary>
+        /// Gets or sets Host Name for binding
+        /// </summary>
+        public string HostName { get; set; }
+
+        /// <inheritdoc />
+        public byte[] CertificateHash { get; set; }
+
+        /// <inheritdoc />
+        public string CertificateStoreName { get; set; }
+
+        /// <inheritdoc />
+        public BindingProtocol BindingProtocol { get; set; }
+
+        /// <inheritdoc />
+        public virtual string BindingInformation
+        {
+            get
+            {
+                return string.Format(@"{0}:{1}:{2}", IpAddress, Port, HostName);
+            }
+        }
+
+        ICustomBindingSettings ICustomBindingSettings.Port(int port)
```

Comment:
Prefix function names with "Set" and change them to generic extension methods around an IBindingSettings interface. [Use this as an example](https://github.com/SharpeRAD/Cake.AWS.S3/blob/master/src/S3/Extensions/Settings/S3SettingsExtensions.cs)